src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -1,243 +1,337 @@
-                           delete root;
-                           root = nullptr;
-                         }
-                       };
+    keyCount++;
 
-    // Check if root needs resizing or cleanup
-    if ((root != nullptr) && (message.length() * 1.5 > lastJsonMessageLength)) {
-      cleanupJSON();
+    if (keyCount > MAX_KEYS) {
+      addLogMove(LOG_LEVEL_ERROR, strformat(F("Warning: More than %d keys in %s"), MAX_KEYS, fileName.c_str()));
     }
 
-    // Resize lastJsonMessageLength if needed
-    if (message.length() * 1.5 > lastJsonMessageLength) {
-      lastJsonMessageLength = message.length() * 1.5;
+    // Process the key and navigate the JSON
+    JsonVariant value = *root;
+    size_t start = 0, end;
+
+    while ((end = key.indexOf('.', start)) != -1) {
+      String part = key.substring(start, end);
+      value = value[part];
+
+      if (value.isNull()) {
+        break; // Key path is invalid
+      }
+      start = end + 1;
     }
 
-    // Allocate memory for root if needed
-    if (root == nullptr) {
-# ifdef USE_SECOND_HEAP
-      HeapSelectIram ephemeral;
-# endif                                                                            // ifdef USE_SECOND_HEAP
-      root = new (std::nothrow) DynamicJsonDocument(lastJsonMessageLength); // Dynamic allocation
+    // Handle the last part of the key
+    if (!value.isNull()) {
+      String lastPart = key.substring(start);
+      value = value[lastPart];
+      successfullyProcessedCount++;
     }
 
-    if (root != nullptr) {
-      // Parse the JSON
-      DeserializationError error = deserializeJson(*root, message);
-
-      if (!error) {
-        int dayEnergy   = (*root)["Body"]["Data"]["DAY_ENERGY"]["Values"]["1"].as<int>();
-        int pac         = (*root)["Body"]["Data"]["PAC"]["Values"]["1"].as<int>();
-        int totalEnergy = (*root)["Body"]["Data"]["TOTAL_ENERGY"]["Values"]["1"].as<int>();
-        int yearEnergy  = (*root)["Body"]["Data"]["YEAR_ENERGY"]["Values"]["1"].as<int>();
-
-        addLog(LOG_LEVEL_INFO, strformat(F("Inverter: Day Energy: %d, PAC: %d, Total Energy: %d, Year Energy: %d"),
-                                         dayEnergy,
-                                         pac,
-                                         totalEnergy,
-                                         yearEnergy));
-        eventQueue.addMove(strformat(F("InverterReply=%d,%d,%d,%d"), dayEnergy, pac, totalEnergy, yearEnergy));
+    // Append the value to the CSV string if it exists
+    if (!value.isNull()) {
+      if (value.is<int>()) {
+        csvOutput += String(value.as<int>());
+      } else if (value.is<float>()) {
+        csvOutput += String(value.as<float>());
+      } else if (value.is<const char *>()) {
+        csvOutput += String(value.as<const char *>());
       } else {
-        Serial.println("Failed to parse JSON");
+        csvOutput += "unknown";
       }
+    } else {
+      csvOutput += "null"; // Indicate missing value
```

Comment:
This is sent as an argument for an event.
However we cannot match a string type.
Wouldn't it make more sense to just not add anything?
So your string then becomes something like `12,34,,78`
instead of `12,34,null,78`

Reply:
Great, i accidentally deleted my replies. I hope you had the chance to read it. Somebody is stressing me to go outside 🤷‍♂️ I rewrite it when i am back....